Выполнена задача спринта №22

Jahnta opened the pull request:
Реализовано создание плейлистов, их отображение и добавление треков

Jahnta committed:
Реализовано создание плейлистов, их отображение и добавление треков

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/data/db/FavouritesRepositoryImpl.kt:
```diff
@@ -32,6 +33,9 @@ class FavouritesRepositoryImpl(
         favouriteTracks.contains(track.trackId)
     }
 
+
```
пустые строки могут улучшать читаемость кода, но не в этом случае. Советую избавляться от дублирующихся пустых строк, они увеличивают размер кода тем самым ухудшая читаемость

vlad-zotov commented on app/build.gradle:
```diff
@@ -35,6 +36,7 @@ android {
 
     buildFeatures {
         viewBinding = true
+        dataBinding = true
```
Привет! Как я вижу ты не пользуешься датабиндингами в проекте, поэтому можешь их отключить чтобы не путаться =)

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/di/RepositoryModule.kt:
```diff
@@ -15,8 +19,14 @@ val repositoryModule = module {
     }
+
+    single<PlaylistRepository> {
+        PlaylistRepositoryImpl(get(), get(), get())
```
советую именовать аргументы конструкторов и функций, чтобы было понятнее что туда передается

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/media/PlaylistsAdapter.kt:
```diff
@@ -0,0 +1,50 @@
+import com.practicum.playlistmaker.databinding.PlaylistViewBinding
+import com.practicum.playlistmaker.domain.media.model.Playlist
+
+class PlaylistsAdapter(private var onClicked: ((Playlist) -> Unit)? = null) :
```
можно сделать поле лиснера `val`. Сомневаюсь, что ты его изменяешь извне. Если этот так, то не советую так делать. Лучше передавать один объект слушателя для каждого объекта адаптера.

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/media/fragment/MediaFavouritesFragment.kt:
```diff
@@ -47,8 +48,9 @@ class MediaFavouritesFragment : Fragment() {
-                PlayerActivity.newIntent(requireContext(), it)
-                    .apply { startActivity(this) }
+                val bundle = Bundle()
+                bundle.putParcelable("track", it)
```
ключи, по которым ты передаешь значения удобнее хранить в константах. Так у тебя будет меньше шанс запутаться

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/player/PlaylistBottomsheetViewHolder.kt:
```diff
@@ -0,0 +1,34 @@
+        binding.tvTitle.text = playlist.playlistTitle
+        val innerNumber = playlist.playlistTracksCount.toString()
+        val text = when {
+            innerNumber.toInt() % 10 == 1 && innerNumber.toInt() % 100 != 11 -> " трек"
```
для склонения слов в андроид есть механизм `plurals` Вот здесь можешь почитать подробнее https://developer.alexanderklimov.ru/android/theory/plurals.php

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/player/activity/PlayerActivity.kt:
```diff
@@ -78,6 +98,89 @@ class PlayerActivity : AppCompatActivity() {
+        var trackIsAdded = false
+        lifecycleScope.launch {
+            viewModel.addTrackToPlaylist(track, playlist)
+            delay(300)
```
не совсем понял зачем нужна эта задержка. По идее она здесь не нужна, а если это небходимо, то советую вынести значение в константу с именем, которое бы передавало смысл назначения этой константы.

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/player/fragment/PlayerFragment.kt:
```diff
@@ -0,0 +1,210 @@
+        binding.trackName.text = track?.trackName
+        binding.artistName.text = track?.artistName
+        binding.trackDuration.text =
+            SimpleDateFormat("mm:ss", Locale.getDefault()).format(track?.trackTimeMillis)
```
Рекомендую весь код, связанный с форматированием времени и дат вынести в отдельный класс-утилиту `object DateTimeUtil`. Это позволяет избавиться от дублирования кода, так же удобно следить за всеми форматами дат и времени в проекте. Ещё этот класс будет легко перенести в другой проект.

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/player/fragment/PlayerFragment.kt:
```diff
@@ -0,0 +1,210 @@
+        var trackIsAdded = false
+        lifecycleScope.launch {
+            viewModel.addTrackToPlaylist(track, playlist)
+            delay(300)
```
здесь тоже не совсем понимаю предназначение задержки

vlad-zotov commented on app/src/main/res/drawable/playlist_placeholder_image.xml:
```diff
@@ -0,0 +1,9 @@
+    android:viewportHeight="80">
+  <path
+      android:pathData="M67.167,21.167V33.625C67.167,33.625 58.875,33.667 58.833,33.625V21.167H46.333C46.333,21.167 46.375,12.875 46.333,12.833H58.833V0.333H67.167V12.833H79.667V21.167H67.167ZM54.667,37.833V25.333H42.167V12.833H8.833C4.25,12.833 0.5,16.583 0.5,21.167V71.167C0.5,75.75 4.25,79.5 8.833,79.5H58.833C63.417,79.5 67.167,75.75 67.167,71.167V37.833H54.667ZM8.833,71.167L21.333,54.5L29.667, ...
+      android:fillColor="#AEAFB4"/>
```
цвета следует выносить в ресурсы

vlad-zotov commented on app/src/main/res/layout/fragment_new_playlist.xml:
```diff
@@ -0,0 +1,135 @@
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:layout_marginStart="12dp"
```
размеры и отступы следует выносить в ресурсы

vlad-zotov commented on app/src/main/res/layout/fragment_new_playlist.xml:
```diff
@@ -0,0 +1,135 @@
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="12dp"
```
размеры и отступы следует выносить в ресурсы

vlad-zotov commented on app/src/main/res/layout/fragment_new_playlist.xml:
```diff
@@ -0,0 +1,135 @@
+        android:hint="@string/playlist_title_input_layout"
+        app:boxBackgroundColor="#00FFFFFF"
+        app:boxBackgroundMode="outline"
+        app:boxCornerRadiusBottomEnd="4dp"
```
размеры и отступы следует выносить в ресурсы. Здесь и ниже ничего не вынесено. Удобно иметь в проекте ресурсы с заготовленными размерами, которые можно переиспользовать в проекте

vlad-zotov commented on app/src/main/res/layout/fragment_new_playlist.xml:
```diff
@@ -0,0 +1,135 @@
+        android:layout_marginTop="32dp"
+        android:layout_marginEnd="12dp"
+        android:hint="@string/playlist_desc_input_layout"
+        app:boxBackgroundColor="#00FFFFFF"
```
цвета следует выносить в ресусры

Jahnta committed:
Исправлены ошибки:
- клавиатура больше не закрывает поле ввода
- исправлен дизайн Bottom Sheet
- добавлено слово "треков" на экране с плейлистами

Jahnta committed:
Исправлено поведение BottomSheet

Исправлено поведение BottomSheet

vlad-zotov commented on app/src/main/java/com/practicum/playlistmaker/ui/media/PlaylistsAdapter.kt:
```diff
@@ -18,7 +18,15 @@ class PlaylistsAdapter(private var onClicked: ((Playlist) -> Unit)? = null) :
             binding.playlistTitle.text = playlist.playlistTitle
-            binding.playlistTracksCount.text = playlist.playlistTracksCount.toString()
+            val innerNumber = playlist.playlistTracksCount.toString()
+            val text = when {
```
советую изучить механизм `plurals` https://developer.android.com/guide/topics/resources/string-resource#Plurals Твой подход неплох, но он разбивается на моменте, когда в приложении требуется завезти локализацию

Jahnta committed:
Исправлен цвет Bottom Sheet в темной теме
